chapter_recurrent-neural-networks/rnn-scratch.md

```diff
@@ -342,9 +342,14 @@ Y.shape, len(new_state), new_state[0].shape
 
 ```{.python .input}
 #@tab tensorflow
+# defining tensorflow training strategy
+device_name = d2l.try_gpu()._device_name
+strategy = tf.distribute.OneDeviceStrategy(device_name)
```

Comment:
Should we just initialize this inside the training function so that we can remove the additional `strategy` parameter as well?

Reply:
@terrytangyuan I need to initialize it here because I need to call the model inside strategy.scope().
And later I'm using this instance of the model in the parameter of the training function like this, train_ch8(model, train_iter, vocab, num_hiddens, lr, num_epochs, strategy).
So, I've already defined the strategy above in my code and I'm using it in the training function.